pkg/provider/resource_cluster.go

```diff
@@ -652,11 +675,42 @@ func makeClusterForCreate(clusterResource ClusterResourceModel) models.Cluster {
 		}
 	}
 
+	if strings.Contains(clusterResource.CloudProvider.ValueString(), "bah") {
+		pids, _ := c.client.GetPeAllowedPrincipalIds(ctx, clusterResource.ProjectId, clusterResource.CloudProvider.ValueString(), clusterResource.Region.ValueString())
+
+		if clusterResource.PeAllowedPrincipalIds.IsNull() {
+			x, _ := clusterResource.PeAllowedPrincipalIds.ToListValue(ctx)
+			var plist []string
+			for _, v := range x.Elements() {
+				plist = append(plist, v.String())
+			}
+			cluster.PeAllowedPrincipalIds = &plist
+
+		} else if len(pids.Data) != 0 {
+			cluster.PeAllowedPrincipalIds = utils.ToPointer(pids.Data)
+		}
+
+		if clusterResource.CloudProvider.ValueString() == "bah:gcp" {
+			if clusterResource.ServiceAccountIds.IsNull() {
+				x, _ := clusterResource.ServiceAccountIds.ToListValue(ctx)
+				var slist []string
+				for _, v := range x.Elements() {
+					slist = append(slist, v.String())
+				}
+				cluster.ServiceAccountIds = &slist
+
+			} else {
+				sids, _ := c.client.GetServiceAccountIds(ctx, clusterResource.ProjectId, clusterResource.CloudProvider.ValueString(), clusterResource.Region.ValueString())
+				cluster.ServiceAccountIds = utils.ToPointer(sids.Data)
+			}
+		}
+	}
+
 	return cluster
 }
 
-func makeClusterFoUpdate(clusterResource ClusterResourceModel) *models.Cluster {
-	cluster := makeClusterForCreate(clusterResource)
+func (c *clusterResource) makeClusterFoUpdate(ctx context.Context, clusterResource ClusterResourceModel) *models.Cluster {
```

Comment:
Do similar to this, as explained on `makeClusterForCreate`

Reply:
I'm not so sure if we need any change here. 
It's already calling the `makeClusterForCreate` method in the next line. And that makeClusterForCreate method has the BAH specific changes. 
If I'm missing something, can you provide more information, please? Thank you.